### PR TITLE
chore: use pnpm to manage dependencies (closes #12635)

### PR DESCRIPTION
**Pull request by thmsobrmlr**
**Release notes: Let's make sure to include Jacob here.**

Continuing the excellent work of @jacobwgillespie at https://github.com/PostHog/posthog/pull/12635 here, since apparently GitHub does not allow edits by maintainers if the repo and the fork are located in organization accounts.

Associated Handbook update here: https://github.com/PostHog/posthog.com/pull/4865

> ## Problem
> Followup from #12520 (see there for more context), this PR replaces Yarn 1 with pnpm. Yarn 1 is semi-unmaintained and has caused issues in `docker build`.
> 
> ## Changes
> This PR switches to use `pnpm` instead of `yarn`. The highlights are:
> 
> * All yarn commands are updated to their pnpm equivalents
> * GitHub Actions workflows and the Dockerfile use pnpm now
> * Some additional dependencies have been added to `package.json` if they previously were only available due to hoisting.
> 
> ## How did you test this code?
> Installed locally + ran `docker build`. Will need CI to run the tests.

### CODE REVIEW

**Comment by jacobwgillespie** [comment]
Nice, thanks for picking this up, I haven't had many spare cycles to finish the PR lately 🙌

File: .github/workflows/ci-frontend.yml [hunk 1]
```diff
@@ -90,27 +87,24 @@ jobs:
 
         steps:
             - uses: actions/checkout@v3
+
+            - name: Install pnpm
+              uses: pnpm/action-setup@v2.2.4
```
Comment (Twixes) [inline .github/workflows/ci-frontend.yml:92]: Why not pin it to v2 or v2.2?

File: frontend/src/lib/components/Popup/Popup.tsx [hunk 2]
```diff
@@ -70,168 +70,166 @@ let nestedPopupReceivedClick = false
...
-            closeParentPopupOnClickInside = false,
-            style,
-            getPopupContainer,
-            showArrow,
-        },
-        ref
-    ): JSX.Element => {
-        const popupId = useMemo(() => uniqueMemoizedIndex++, [])
-        const parentPopupId = useContext(PopupContext)
+export const Popup = React.forwardRef<HTMLDivElement, PopupProps>(function Popup(
```
Comment (Twixes) [inline frontend/src/lib/components/Popup/Popup.tsx:73]: For the inner pre-forwardRef component we often follow a `ComponentInternal` ('PopupInternal` here) style – it's clearer in stack traces

**Review by Twixes — APPROVED** [review]
It works!

File: .github/workflows/ci-frontend.yml (hunk 1 above)
Comment (thmsobrmlr) [reply .github/workflows/ci-frontend.yml:92]: Good catch! Will pin to v2 as well.

File: frontend/src/lib/components/Popup/Popup.tsx (hunk 2 above)
Comment (thmsobrmlr) [reply frontend/src/lib/components/Popup/Popup.tsx]: Thanks, that makes sense! Updated.

**Comment by thmsobrmlr** [comment]
> Nice, thanks for picking this up, I haven't had many spare cycles to finish the PR lately 🙌

Thanks for your awsome work at https://github.com/PostHog/posthog/pull/12635, @jacobwgillespie! We probably wouldn't have made the switch yet without your contribution and are really happy to have it. I'll make sure to include you in the next release notes :)

**Comment by jacobwgillespie** [comment]
Awesome, happy to see it merge! 🚀

**Pull request merged**